doc/advanced_configuration.rst

```diff
@@ -142,6 +143,29 @@ If you so desire you can implement your own sorting key. It will be
 provided the relative paths to `conf.py` of each sub gallery folder.
 
 
+.. _within_gallery_order:
+
+Sorting gallery examples
+========================
+
+Within a given gallery (sub)section, the example files by default are ordered
```

Comment:
Do we have any short description in the docs that just explains that ordering in general is handled by "sorting keys" which are a way of defining a custom sorting behavior with the `sorted` function? Maybe that's too much info but I was confused the first time I saw sort keys being used.